plugins/router/f5/f5.go

```diff
@@ -663,6 +683,99 @@ func (f5 *f5LTM) ensureVserverHasIRule(vserverName, iRuleName string) error {
 	return nil
 }
 
+// checkPartitionPathExists checks if the partition path exists.
+func (f5 *f5LTM) checkPartitionPathExists(pathName string) (bool, error) {
+	glog.V(4).Infof("Checking if partition path %q exists...", pathName)
+
+	// Ensure pathname is absolute.
+	if !path.IsAbs(pathName) {
+		pathName = path.Join("/", pathName)
+	}
+
+	// F5 iControl REST API expects / characters in the path to be
+	// escaped as ~.
+	uri := fmt.Sprintf("https://%s/mgmt/tm/sys/folder/%s",
+		f5.host, strings.Replace(pathName, "/", "~", -1))
+
+	err := f5.get(uri, nil)
+	if err != nil {
+		if err.(F5Error).httpStatusCode != 404 {
+			glog.Errorf("partition path %q error: %v", pathName, err)
+			return false, err
+		}
+
+		//  404 is ok means that the path doesn't exist == !err.
+		return false, nil
+	}
+
+	glog.V(4).Infof("Partition path %q exists.", pathName)
+	return true, nil
+}
+
+// addPartitionPath adds a new partition path to the folder heirarchy.
+func (f5 *f5LTM) addPartitionPath(pathName string) (bool, error) {
+	glog.V(4).Infof("Creating partition path %q ...", pathName)
+
+	uri := fmt.Sprintf("https://%s/mgmt/tm/sys/folder", f5.host)
+
+	payload := f5AddPartitionPathPayload{Name: pathName}
+	err := f5.post(uri, payload, nil)
+	if err != nil {
+		if err.(F5Error).httpStatusCode != 409 {
+			glog.Errorf("Error adding partition path %q error: %v", pathName, err)
+			return false, err
+		}
+
+		// If the path already exists, don't return an error.
+		glog.Warningf("Partition path %q not added as it already exists.", pathName)
+		return false, nil
+	}
+
+	return true, nil
+}
+
+// ensurePartitionPathExists checks whether the specified partition path
+// hierarchy exists and creates it if it does not.
+func (f5 *f5LTM) ensurePartitionPathExists(pathName string) error {
+	glog.V(4).Infof("Ensuring partition path %s exists...", pathName)
+
+	exists, err := f5.checkPartitionPathExists(pathName)
+	if err != nil {
+		return err
+	}
+
+	if exists {
+		glog.V(4).Infof("Partition path %s already exists; nothing to do.", pathName)
+		return nil
+	}
+
+	// We have to loop thru the path heirarchy and add components
+	// individually if they don't exist.
+
+	// Get path components - we need to remove the leading slash from
+	// the pathname (make it absolute if it is not already and remove
+	// the leading slash).
+	p := "/"
+	pathComponents := strings.Split(path.Join("/", pathName)[1:], "/")
+	for _, v := range pathComponents {
+		p = path.Join(p, v)
+
+		if _, err := f5.checkPartitionPathExists(p); err != nil {
+			//  PartitionPath components exists, go to next.
+			continue
```

Comment:
I'm confused.  Shouldn't we continue iff the first return value of `f5.checkPartitionPathExists(p)` is true and log or bail if `err != nil`?